### PR TITLE
Deprecate the getSearchablePages() method

### PR DESCRIPTION
**Pull request by aschempp**
fixes https://github.com/contao/contao/pull/3150#issuecomment-874158385

### CODE REVIEW

**Review by fritzmg — CHANGES_REQUESTED** [review]
This should be a service and not a private function, otherwise we cannot retrieve the searchable pages elsewhere.

**Comment by aschempp** [comment]
Why would you need to retrieve the searchable pages elsewhere? Also, I'll rename the method because these are not searchable pages 😂

**Review by Toflar — COMMENTED** [review]
Implementation is correct, just requires the CI chain to pass. And no, it does not need to be a service. Nobody needs to access this list :)

**Review by fritzmg — DISMISSED** [review]

File: core-bundle/src/Controller/SitemapController.php [hunk 1]
```diff
@@ -110,4 +122,55 @@ private function callLegacyHook(PageModel $rootPage, array $pages): array
 
         return $pages;
     }
+
+    private function getPageAndArticleUrls(int $pid): array
```
Comment (Toflar) [inline core-bundle/src/Controller/SitemapController.php]: Shouldn't we use individual `SitemapEvent` listeners for articles and pages?

**Comment by leofeyer** [comment]
Looks good to me. 👍🏻 Will need to fix the CI chain and answer https://github.com/contao/contao/pull/3164#discussion_r665501221 though.

File: core-bundle/src/Controller/SitemapController.php (hunk 1 above)
Comment (aschempp) [reply core-bundle/src/Controller/SitemapController.php]: This would cause multiple issues, e.g. the pages would no longer be in the legacy hook.

**Review by Toflar — DISMISSED** [review]

**Review by Toflar — DISMISSED** [review]

**Review by leofeyer — APPROVED** [review]

**Review by fritzmg — APPROVED** [review]

**Pull request merged**